--- a/collectstats.pl	2005-01-21 08:57:30.975582527 +0200
+++ a/collectstats.pl	2006-04-24 23:04:25.847159729 +0300
@@ -48,7 +48,7 @@
 if (chdir("graphs")) {
     unlink <./*.gif>;
     unlink <./*.png>;
-    chdir("..");
+    chdir("/usr/share/bugzilla");
 }
 
 GetVersionTable();
@@ -96,8 +96,8 @@
     trick_taint($perl);
 
     # Generate a static RDF file containing the default view of the duplicates data.
-    open(CGI, "$perl -T duplicates.cgi |")
-        || die "can't fork duplicates.cgi: $!";
+    open(CGI, "$perl -T htdocs/duplicates.cgi |")
+        || die "can't fork htdocs/duplicates.cgi: $!";
     open(RDF, ">$datadir/duplicates.tmp")
         || die "can't write to $datadir/duplicates.tmp: $!";
     my $headers_done = 0;
--- bugzilla-3.0/Bugzilla/Constants.pm~	2007-05-10 01:47:17.000000000 +0300
+++ bugzilla-3.0/Bugzilla/Constants.pm	2007-06-21 23:57:48.044744852 +0300
@@ -407,10 +407,10 @@
     my ($project, $localconfig, $datadir);
     if ($ENV{'PROJECT'} && $ENV{'PROJECT'} =~ /^(\w+)$/) {
         $project = $1;
-        $localconfig = "localconfig.$project";
+        $localconfig = "localconfig.$project.pl";
         $datadir = "data/$project";
     } else {
-        $localconfig = "localconfig";
+        $localconfig = "localconfig.pl";
         $datadir = "data";
     }
 
@@ -420,13 +420,13 @@
         'libpath'     => $libpath,
         # If you put the libraries in a different location than the CGIs,
         # make sure this still points to the CGIs.
-        'cgi_path'    => $libpath,
-        'templatedir' => "$libpath/template",
+        'cgi_path'    => '/usr/share/bugzilla/htdocs',
+        'templatedir' => "/usr/share/bugzilla/template",
         'project'     => $project,
-        'localconfig' => "$libpath/$localconfig",
-        'datadir'     => "$libpath/$datadir",
-        'attachdir'   => "$libpath/$datadir/attachments",
-        'skinsdir'    => "$libpath/skins/contrib",
+        'localconfig' => "/etc/webapps/bugzilla/$localconfig",
+        'datadir'     => "/var/lib/bugzilla/$datadir",
+        'attachdir'   => "/var/lib/bugzilla/$datadir/attachments",
+        'skinsdir'    => "/usr/share/bugzilla/htdocs/skins/contrib",
         # $webdotdir must be in the webtree somewhere. Even if you use a 
         # local dot, we output images to there. Also, if $webdotdir is 
         # not relative to the bugzilla root directory, you'll need to 
--- bugzilla-3.0/Bugzilla/Install/Filesystem.pm	2006-12-09 13:58:14.000000000 +0200
+++ bugzilla-3.0/Bugzilla/Install/Filesystem.pm	2007-06-22 00:08:36.399403065 +0300
@@ -99,21 +99,6 @@
 
     # --- FILE PERMISSIONS (Non-created files) --- #
     my %files = (
-        '*'               => { perms => $ws_readable },
-        '*.cgi'           => { perms => $ws_executable },
-        'whineatnews.pl'  => { perms => $ws_executable },
-        'collectstats.pl' => { perms => $ws_executable },
-        'checksetup.pl'   => { perms => $owner_executable },
-        'importxml.pl'    => { perms => $ws_executable },
-        'runtests.pl'     => { perms => $owner_executable },
-        'testserver.pl'   => { perms => $ws_executable },
-        'whine.pl'        => { perms => $ws_executable },
-        'customfield.pl'  => { perms => $owner_executable },
-        'email_in.pl'     => { perms => $ws_executable },
-
-        'docs/makedocs.pl'   => { perms => $owner_executable },
-        'docs/rel_notes.txt' => { perms => $ws_readable },
-        'docs/README.docs'   => { perms => $owner_readable },
         "$datadir/bugzilla-update.xml" => { perms => $ws_writeable },
         "$datadir/params" => { perms => $ws_writeable },
         "$datadir/mailer.testfile" => { perms => $ws_writeable },
@@ -123,8 +108,6 @@
     # recurse through. These are directories we didn't create
     # in checkesetup.pl.
     my %non_recurse_dirs = (
-        '.'  => $ws_dir_readable,
-        docs => $ws_dir_readable,
     );
 
     # This sets the permissions for each item inside each of these 
@@ -138,29 +121,13 @@
          $attachdir         => { files => $ws_writeable,
                                   dirs => $ws_dir_writeable },
          $webdotdir         => { files => $ws_writeable,
                                   dirs => $ws_dir_writeable },
-         graphs             => { files => $ws_writeable,
-                                  dirs => $ws_dir_writeable },
 
          # Readable directories
          "$datadir/mining"     => { files => $ws_readable,
                                      dirs => $ws_dir_readable },
          "$datadir/duplicates" => { files => $ws_readable,
                                      dirs => $ws_dir_readable },
-         "$libdir/Bugzilla"    => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         $templatedir          => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         images                => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         css                   => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         js                    => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         skins                 => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         t                     => { files => $owner_readable,
-                                     dirs => $owner_dir_readable },
          'docs/html'           => { files => $ws_readable,
                                      dirs => $ws_dir_readable },
          'docs/pdf'            => { files => $ws_readable,
@@ -186,7 +153,7 @@
         "$datadir/duplicates"   => $ws_dir_readable,
         $attachdir              => $ws_dir_writeable,
         $extensionsdir          => $ws_dir_readable,
-        graphs                  => $ws_dir_writeable,
+        "$datadir/graphs"       => $ws_dir_writeable,
         $webdotdir              => $ws_dir_writeable,
         'skins/custom'          => $ws_dir_readable,
         'skins/contrib'         => $ws_dir_readable,
