diff -burNp bugzilla-3.3.3/Bugzilla/Constants.pm bugzilla-3.3.3-dud/Bugzilla/Constants.pm
--- bugzilla-3.3.3/Bugzilla/Constants.pm	2009-02-03 11:03:46.000000000 +0100
+++ bugzilla-3.3.3-dud/Bugzilla/Constants.pm	2009-03-18 16:41:34.475759477 +0100
@@ -482,10 +482,10 @@ sub bz_locations {
     my ($project, $localconfig, $datadir);
     if ($ENV{'PROJECT'} && $ENV{'PROJECT'} =~ /^(\w+)$/) {
         $project = $1;
-        $localconfig = "localconfig.$project";
+        $localconfig = "localconfig.$project.pl";
         $datadir = "data/$project";
     } else {
-        $localconfig = "localconfig";
+        $localconfig = "localconfig.pl";
         $datadir = "data";
     }
 
@@ -496,13 +496,13 @@ sub bz_locations {
         'ext_libpath' => "$libpath/lib",
         # If you put the libraries in a different location than the CGIs,
         # make sure this still points to the CGIs.
-        'cgi_path'    => $libpath,
-        'templatedir' => "$libpath/template",
+        'cgi_path'    => '/usr/share/bugzilla/htdocs',
+        'templatedir' => "/usr/share/bugzilla/template",
         'project'     => $project,
-        'localconfig' => "$libpath/$localconfig",
-        'datadir'     => "$libpath/$datadir",
-        'attachdir'   => "$libpath/$datadir/attachments",
-        'skinsdir'    => "$libpath/skins",
+        'localconfig' => "/etc/webapps/bugzilla/$localconfig",
+        'datadir'     => "/var/lib/bugzilla/$datadir",
+        'attachdir'   => "/var/lib/bugzilla/$datadir/attachments",
+        'skinsdir'    => "/usr/share/bugzilla/htdocs/skins",
         # $webdotdir must be in the web server's tree somewhere. Even if you use a 
         # local dot, we output images to there. Also, if $webdotdir is 
         # not relative to the bugzilla root directory, you'll need to 
diff -burNp bugzilla-3.3.3/Bugzilla/Install/Filesystem.pm bugzilla-3.3.3-dud/Bugzilla/Install/Filesystem.pm
--- bugzilla-3.3.3/Bugzilla/Install/Filesystem.pm	2009-01-23 22:34:42.000000000 +0100
+++ bugzilla-3.3.3-dud/Bugzilla/Install/Filesystem.pm	2009-03-18 16:44:33.952669848 +0100
@@ -103,27 +103,6 @@ sub FILESYSTEM {
 
     # --- FILE PERMISSIONS (Non-created files) --- #
     my %files = (
-        '*'               => { perms => $ws_readable },
-        '*.cgi'           => { perms => $ws_executable },
-        'whineatnews.pl'  => { perms => $ws_executable },
-        'collectstats.pl' => { perms => $ws_executable },
-        'checksetup.pl'   => { perms => $owner_executable },
-        'importxml.pl'    => { perms => $ws_executable },
-        'runtests.pl'     => { perms => $owner_executable },
-        'testserver.pl'   => { perms => $ws_executable },
-        'whine.pl'        => { perms => $ws_executable },
-        'customfield.pl'  => { perms => $owner_executable },
-        'email_in.pl'     => { perms => $ws_executable },
-        'sanitycheck.pl'  => { perms => $ws_executable },
-        'jobqueue.pl'     => { perms => $owner_executable },
-        'install-module.pl' => { perms => $owner_executable },
-
-        "$localconfig.old" => { perms => $owner_readable },
-
-        'docs/makedocs.pl'   => { perms => $owner_executable },
-        'docs/style.css'       => { perms => $ws_readable },
-        'docs/*/rel_notes.txt' => { perms => $ws_readable },
-        'docs/*/README.docs'   => { perms => $owner_readable },
         "$datadir/bugzilla-update.xml" => { perms => $ws_writeable },
         "$datadir/params" => { perms => $ws_writeable },
         "$datadir/mailer.testfile" => { perms => $ws_writeable },
@@ -133,8 +112,6 @@ sub FILESYSTEM {
     # recurse through. These are directories we didn't create
     # in checkesetup.pl.
     my %non_recurse_dirs = (
-        '.'  => $ws_dir_readable,
-        docs => $ws_dir_readable,
     );
 
     # This sets the permissions for each item inside each of these 
@@ -149,32 +126,12 @@ sub FILESYSTEM {
                                   dirs => $ws_dir_writeable },
          $webdotdir         => { files => $ws_writeable,
                                   dirs => $ws_dir_writeable },
-         graphs             => { files => $ws_writeable,
-                                  dirs => $ws_dir_writeable },
 
          # Readable directories
          "$datadir/mining"     => { files => $ws_readable,
                                      dirs => $ws_dir_readable },
          "$datadir/duplicates" => { files => $ws_readable,
                                      dirs => $ws_dir_readable },
-         "$libdir/Bugzilla"    => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         $extlib               => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         $templatedir          => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         $extensionsdir        => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         images                => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         css                   => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         js                    => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         $skinsdir             => { files => $ws_readable,
-                                     dirs => $ws_dir_readable },
-         t                     => { files => $owner_readable,
-                                     dirs => $owner_dir_readable },
          'docs/*/html'         => { files => $ws_readable,
                                      dirs => $ws_dir_readable },
          'docs/*/pdf'          => { files => $ws_readable,
@@ -199,7 +156,6 @@ sub FILESYSTEM {
         "$datadir/duplicates"   => $ws_dir_readable,
         $attachdir              => $ws_dir_writeable,
         $extensionsdir          => $ws_dir_readable,
-        graphs                  => $ws_dir_writeable,
         $webdotdir              => $ws_dir_writeable,
         "$skinsdir/custom"      => $ws_dir_readable,
         "$skinsdir/contrib"     => $ws_dir_readable,
diff -burNp bugzilla-3.3.3/collectstats.pl bugzilla-3.3.3-dud/collectstats.pl
--- bugzilla-3.3.3/collectstats.pl	2008-12-16 22:16:28.000000000 +0100
+++ bugzilla-3.3.3-dud/collectstats.pl	2009-03-18 16:38:32.469101656 +0100
@@ -56,7 +56,7 @@ if (chdir("graphs")) {
     unlink <./*.gif>;
     unlink <./*.png>;
     # chdir("..") doesn't work if graphs is a symlink, see bug 429378
-    chdir($cwd);
+    chdir("/usr/share/bugzilla");
 }
 
 my $dbh = Bugzilla->switch_to_shadow_db();
